radixdlt/src/main/java/com/radixdlt/consensus/PendingVotes.java

```diff
@@ -41,26 +81,74 @@ public PendingVotes(Hasher hasher) {
 
 	/**
 	 * Inserts a vote for a given vertex, attempting to form a quorum certificate for that vertex.
-	 *
+	 * <p>
 	 * A QC will only be formed if permitted by the {@link ValidatorSet}.
+	 *
 	 * @param vote The vote to be inserted
 	 * @return The generated QC, if any
 	 */
 	public Optional<QuorumCertificate> insertVote(Vote vote, ValidatorSet validatorSet) {
-		final Hash voteHash = hasher.hash(vote.getVoteData());
-
-		ECDSASignature signature = vote.getSignature().orElseThrow(() -> new IllegalArgumentException("vote is missing signature"));
-		ValidationState validationState = votes.computeIfAbsent(voteHash, validatorSet::newValidationState);
-
-		// try to form a QC with the added signature according to the requirements
-		if (!validationState.addSignature(vote.getAuthor(), signature)) {
-			// if no QC could be formed, update pending and return nothing
-			return Optional.empty();
-		} else {
-			// if QC could be formed, remove pending and return formed QC
-			votes.remove(voteHash);
-			QuorumCertificate qc = new QuorumCertificate(vote.getVoteData(), validationState.signatures());
-			return Optional.of(qc);
+		final ECPublicKey voteAuthor = vote.getAuthor();
+		final VoteData voteData = vote.getVoteData();
+		final Hash voteHash = this.hasher.hash(voteData);
+		final ECDSASignature signature = vote.getSignature().orElseThrow(() -> new IllegalArgumentException("vote is missing signature"));
+		// Only process for valid validators and signatures
+		if (validatorSet.containsKey(voteAuthor) && voteAuthor.verify(voteHash, signature)) {

```

Comment:
Can we add logging on invalid validators/signatures rather than silently failing?